content/source/layouts/layout.erb

```diff
@@ -6,6 +6,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta name="description" content="<%= description_for(current_page) %>">
 
+    <link rel="canonical" href="<%= File.join(base_url, current_page.url) %>" />
```

Comment:
Is there anyway to override `base_url`? Does it change when people run the site locally or anything? Curious if it would be better to just hardcode this to `https://www.terraform.io/` here. Otherwise though, LGTM

Reply:
I made sure to check! That's [hardcoded in content/config.rb](https://github.com/hashicorp/terraform-website/blob/master/content/config.rb#L1), and doesn't change in preview builds. So you'd have to mess with it on purpose. I checked the source of a preview build, too: 

![Screen Shot 2019-08-30 at 11 14 20 AM](https://user-images.githubusercontent.com/484309/64043458-5ad46280-cb19-11e9-8aa0-945d862667a2.png)

Reply:
This maybe beyond the scope of this PR, but we should be picking a single URL for `/index.html` pages. In the example above, both of these URLs work:

https://www.terraform.io/docs/cloud/
https://www.terraform.io/docs/cloud/index.html

We need to choose one and then redirect the other. If we're going to choose to leave of the `/index.html` the code above needs to trim it from `current_page.url`

Reply:
Right now it does consolidate on a single URL. If you go to the page with the bare /, it'll say index.html is the canonical one. That matches with what most of our links do (for some historical reason, we basically always link to index.html). 

Legit to disagree with which one should be the "official" path, but yeah, that's a conversation for the next web platform, not for this.